src/utils.cpp

```diff
@@ -91,15 +104,97 @@ std::string Logger::dumpPrefixLens() {
     std::string s = "";
     for(size_t i = 0; i < _nrules; ++i) {
         if (_state.prefix_lens[i] > 0) {
-            s += std::to_string(i);
+            s += sizet_tostring(i);
             s += ":";
-            s += std::to_string(_state.prefix_lens[i]);
+            s += sizet_tostring(_state.prefix_lens[i]);
             s += ";";
         }
     }
     return s;
 }
 
+// Custom getline for Windows support
+signed long long m_getline(char** lineptr, size_t* n, FILE* stream)
```

Comment:
Oof do we need custom functions or can we use a Microsoft implementation (a la https://social.msdn.microsoft.com/Forums/vstudio/en-US/94bc15ac-1c91-4252-977e-ea010c3e9d5f/the-usage-of-getline?forum=vclanguage)

Reply:
strsep and strdup were simple enough that I could write my own versions quickly. Using Microsoft's implementation of getline would work, but then we'd have to use C++ code in rulelib.cpp, which is all C at the moment (and I thought that'd be kinda ugly). If you think it's better, I can go ahead and do it. My implementation of getline seems to be working fine, however, on Windows, Mac, and Linux.

Reply:
I would really really prefer to use library functions if possible. Since you moved rulelib.cpp to be compiled by g++ anyways, I think we should go ahead and use the Microsoft implementations for all of these functions. We can wrap them in `#ifdef`s to tell if we're on Windows, no?